File: 704-binary-search/704-binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left = 0
        right = len(nums)
        
        while(left< right):
            mid = (int)(left + (right - left)/2)
            
            
            n = nums[mid]
            
            if (n > target):
                right = mid
            elif (n < target):
                left = mid + 1
            else: 
                return mid
        return -1

File: 704-binary-search/704-binary-search.rs
impl Solution {
    pub fn search(nums: Vec<i32>, target: i32) -> i32 {
        let mut l = 0;
        let mut r = nums.len()-1;
        
        while (l < r){
            let mid = l + (r - l)/2;
            let n = nums[mid];
            
            if n > target{
                r = mid;
            }else if n < target{
                l = mid +1;
            }else{
                return mid as i32;
            }
        }
        
        if nums[r] == target{
            r as i32
        }else{
            -1
        }
    }
}